### ISSUE TITLE
no-undef false positive on class properties/fields.

### ISSUE THREAD

**Issue by jsg2021** [eslint/eslint#8720]
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.0.0
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```json
{
	"extends": "eslint:recommended",
	"parserOptions": {
		"ecmaVersion": 2017,
		"sourceType": "module",
		"ecmaFeatures": {
			"experimentalObjectRestSpread": true,
			"impliedStrict": true,
			"globalReturn": false,
			"jsx": true
		}
	},
	"parser": "babel-eslint"
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
class Foo {
    static bar = true

    baz = 'meh'
}

```

**What did you expect to happen?**
no errors.

**What actually happened? Please include the actual, raw output from ESLint.**
no-undef is treating `bar` and `baz` as errors (assigning new globals?)  This didn't error in eslint 3.x.x

**Comment by kaicataldo** [comment]
This will most likely need to be fixed in babel-eslint. We changed some of our developer-facing APIs and I'm guessing some of the monkeypatching in babel-eslint will need to be updated.

**Comment by not-an-aardvark** [comment]
Closing because class properties are an experimental proposal, and we [don't support](https://github.com/eslint/eslint/blob/597c217f2c0ac5ca4bd1ddad321f2192a88fc40f/README.md#what-about-experimental-features) experimental proposals. If you're encountering this issue when using `babel-eslint`, please follow https://github.com/babel/babel-eslint/issues/487 instead.
